tests/integrationv2/test_hello_retry_requests.py

```diff
@@ -71,3 +71,73 @@ def test_hrr_with_s2n_as_client(managed_process, cipher, provider, curve, protoc
         assert b'Supported Elliptic Groups: X25519:P-256:P-384' in results.stdout
         assert bytes("Shared Elliptic groups: {}".format(server_options.curve).encode('utf-8')) in results.stdout
         assert random_bytes in results.stdout
+
+
+def get_curve_name(curve):
+    if curve.name == "X25519":
+       return "x25519"
+    elif curve.name == "P-256":
+        return "secp256r1"
+    elif curve.name == "P-384":
+        return "secp384r1"
+    else:
+       return None
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", TLS13_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL])
+@pytest.mark.parametrize("curve", ALL_TEST_CURVES, ids=get_parameter_name)
+@pytest.mark.parametrize("protocol", [Protocols.TLS13], ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", ALL_TEST_CERTS, ids=get_parameter_name)
+def test_hrr_with_s2n_as_server(managed_process, cipher, provider, curve, protocol, certificate):
+    port = next(available_ports)
+
+    random_bytes = data_bytes(64)
+    client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        host="localhost",
+        port=port,
+        cipher=cipher,
+        data_to_send=random_bytes,
+        insecure=True,
+        curve=curve,
+        extra_flags = ['-msg', '-curves', 'X448:'+str(curve)],
```

Comment:
What does X448 mean? I thought there were only three curves?

Reply:
Curve X448 is currently not supported in s2n, in this test we use x448 as the first preference to force the test into hello_retry mode.